src/System.Runtime/tests/System/TypeLoadExceptionTests.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Xunit;
+
+namespace System.Tests
+{
+    public static class TypeLoadExceptionTests
+    {
+        private const int COR_E_TYPELOAD = unchecked((int)0x80131522);
+
+        [Fact]
+        public static void Ctor_Empty()
+        {
+            var exception = new TypeLoadException();
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_TYPELOAD, validateMessage: false);
+            Assert.Equal("", exception.TypeName);
+        }
+
+        [Fact]
+        public static void Ctor_String()
+        {
+            string message = "type failed to load";
+            var exception = new TypeLoadException(message);
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_TYPELOAD, message: message);
+            Assert.Equal("", exception.TypeName);
+        }
+
+        [Fact]
+        public static void Ctor_String_Exception()
+        {
+            string message = "type failed to load";
+            var innerException = new Exception("Inner exception");
+            var exception = new TypeLoadException(message, innerException);
+            ExceptionUtility.ValidateExceptionProperties(exception, hResult: COR_E_TYPELOAD, innerException: innerException, message: message);
+            Assert.Equal("", exception.TypeName);
+        }
+    }
```

Comment:
I am not sure whether there is a test verifying TypeName etc when the runtime calls the private ctor. At least I didn't see any case of `Assert.Throws<TypeLoadException>` that seemed to assign it to something to inspect. So my look was very brief..

Reply:
I am focusing just on making sure that there is equivalent CoreFX test coverage for the tests that I am deleting in CoreCLR.

I agree that we can have better coverage for the runtime type loader error handling. The tests that verify exceptions thrown by runtime type loader would belong to CoreCLR. CoreFX does not have the right infrastructure to produce invalid binaries that trigger them.

Opened: https://github.com/dotnet/coreclr/issues/20536